src/EditorFeatures/Core/Implementation/EditAndContinue/EditAndContinueDiagnosticAnalyzer.cs

```diff
@@ -18,6 +18,8 @@ internal sealed class EditAndContinueDiagnosticAnalyzer : DocumentDiagnosticAnal
     {
         private static readonly ImmutableArray<DiagnosticDescriptor> s_supportedDiagnostics = EditAndContinueDiagnosticDescriptors.GetDescriptors();
 
+        private volatile RemoteEditAndContinueServiceProxy _lazyProxy;
```

Comment:
fwiw, i vastly prefer we avoid volatile and instead use any of:

1. `lock`
2. `Lazy<T>`
3. `LazyInitialized`
4. Another mutex type (i.e. Semaphore/SemaphoreSlim)

I honestly find it impossible to get good *consistent* data on how volatile works and how to code completely safely with it.  In practice, unless it's an absolutely massive perf hotspot, it doesn't feel like it is warranted given the conceptual complexity it adds.

Reply:
and, if volatile is necessary, i think it shoudl include docs with a strong argument for why it's the right choice.

Reply:
None of the above is needed here. We don't care about having a unique instance. `volatile` is just a memory barrier, so that the object is fully initialized before it might be read by another thread. There is nothing complex about it.

Reply:
`volatile` is technically not necessary here since jit inserts the memory barrier after every managed reference write, but it is imo good practice to declare it.

Reply:
I'm just gonna allocate a new proxy for each call. That makes is simpler.